synapse/storage/databases/main/purge_events.py

```diff
@@ -102,15 +102,19 @@ def _purge_history_txn(
             (room_id,),
         )
         rows = txn.fetchall()
-        max_depth = max(row[1] for row in rows)
-
-        if max_depth < token.topological:
-            # We need to ensure we don't delete all the events from the database
-            # otherwise we wouldn't be able to send any events (due to not
-            # having any backwards extremities)
-            raise SynapseError(
-                400, "topological_ordering is greater than forward extremeties"
-            )
+        # if we already have no forwards extremities (for example because they were 
+        # cleared out by the `delete_old_current_state_events` background database 
```

Comment:
```suggestion
        # if we already have no forwards extremities (for example because they were
        # cleared out by the `delete_old_current_state_events` background database
```